File: ProblemSolve/344.cpp
/******************
 * Reverse String *
 ******************/

class Solution {
public:
    void reverseString(vector<char>& s) {
        //비어있으면 리턴
        if(s.empty()){
            return;
        }
        
        int len = s.size();
        //swap 함수로 바꾸기
        for(int i=0; i<len/2; i++){
            swap(s[i], s[len -1 -i]);
        }        
    }
};

File: ProblemSolve/344.cs
/******************
 * Reverse String *
 ******************/

public class Solution {
    public void Swap(ref char left, ref char right){
        char temp = left;
        left = right;
        right = temp;
    }
    
    public void ReverseString(char[] s) {
        int left = 0, right = s.Length-1;
        
        while(left < right){
            Swap(ref s[left], ref s[right]);
            left++;
            right--;
        }
        
        return;
    }
}